Michael/symco12/Password Reset

Mkay971 opened the pull request:
this is for symco-12 and symco-63. This is for the password reset form through email from the login page. simply enter in a student or researchers email and they will be sent a link to reset their password, and then directed to login. and there is a fix for the student management nav bar. I also added SYMCO61 to this by accident, but it essentially got rid of the all students label and adds ui for adding a student that doesn't have a label.

Mkay971 committed:
Password Reset

This should complete the password restet form from login.

Mkay971 committed:
Merge branch 'main' into michael/SYMCO12

Mkay971 committed:
bug fix

Bug fix for SYMCO-63

ericharper1 commented on simcon_project/simcon_project/urls.py:
```diff
@@ -27,6 +27,11 @@
+    path('password-reset/', password_reset, name="password_reset"),
+    path('password-reset/done/', views.PasswordResetDoneView.as_view(template_name='registration/password_reset_done.html'),name="password_reset_done"),
+    path('password-reset-confirm/<uidb64>/<token>/', views.PasswordResetConfirmView.as_view(template_name='registration/password_reset_confirm.html'),name="password_reset_confirm"),
+    path('password-reset-complete/', views.PasswordResetCompleteView.as_view(template_name='registration/password_reset_complete.html'),name="password_reset_confirm"),
```
I'm gonna be a jerk here but for the url name, it should be hyphen-case instead of snake_case :)

ericharper1 commented on simcon_project/templates/registration/login.html:
```diff
@@ -4,4 +4,5 @@ <h2>Log In</h2>
   {% csrf_token %}
   {{ form.as_p }}
   <button type="submit">Log In</button>
+    <a href="/password-reset/">Forgot Password?</a>
```
use the url template tag and the url name here instead of hard-coded url

ericharper1 commented on simcon_project/templates/registration/password_reset_complete.html:
```diff
@@ -0,0 +1,6 @@
+<h2>Log In</h2>
+<form method="post">
+    <div class="alert alert-info">Your Password has been reset</div>
+    <a href="/">Sing In Here</a>
```
Typo on line 5

ericharper1 commented on simcon_project/users/views/password_reset.py:
```diff
@@ -0,0 +1,42 @@
+                    # creates the subject and message content for the emails
+                    subject = 'Activate Your Simulated conversations account'
+                    message = 'Hi, \nPlease register here: \nhttp://' + site + '/password-reset-confirm/' \
+                          + uid + '/' + token +'/' '\n'
```
Header and message should be more tailored for password reset

ericharper1 reviewed (changes requested):
Works well and guards against invalid input. Just made some stylistic comments

Mkay971 committed:
made suggested changes

made changes eric suggested.

Mkay971 committed:
Merge branch 'main' into michael/SYMCO12

ericharper1 commented on simcon_project/templates/registration/password_reset_complete.html:
```diff
@@ -0,0 +1,6 @@
+<h2>Log In</h2>
+<form method="post">
+    <div class="alert alert-info">Your Password has been reset</div>
+    <a href="{% url 'login' %}">Sing In Here</a>
```
"sign in here"

ericharper1 reviewed (dismissed):
Awesome!

devin-w-hill commented on simcon_project/users/views/password_reset.py:
```diff
@@ -0,0 +1,42 @@
+            else:
+                messages.error(request, 'Invalid Email', fail_silently=False)
+
+    return render(request, 'registration/password_reset.html', {"form": PassReset()})
```
newline

devin-w-hill commented on simcon_project/users/views/password_reset.py:
```diff
@@ -0,0 +1,42 @@
+                    token=default_token_generator.make_token(user)
+
+                    # creates the subject and message content for the emails
+                    subject = 'Reset Your Simulated conversations account password'
```
C on conversations. nit picky, I know

devin-w-hill commented on simcon_project/users/views/student_management.py:
```diff
@@ -49,40 +49,38 @@ def student_management(request, name="All Students"):
-                        messages.error(request, 'Student already exists',
-                                       fail_silently=False)
+                    # creates the subject and message content for the emails
+                    subject = 'Activate Your Simulated conversations account'
```
C on conversations

devin-w-hill reviewed (dismissed):
Works great. Suggested small changes. Approving anyway, none of my changes should stop this from getting merged. We can fix later.

rosasck reviewed (commented):
but fix devins changes plzz

Mkay971 committed:
added SYMCO-61 on accident

Added SYCMO-61 to this ticket by accident so i just now finished the functionality of it.

Mkay971 committed:
Merge branch 'main' into michael/SYMCO12

# Conflicts:
#	simcon_project/users/views/__init__.py

Mkay971 committed:
removed now unnecessary file

removed now unnecessary file for password reset.

Mkay971 committed:
last fix

Mkay971 committed:
typo
